core/src/main/java/com/linecorp/armeria/client/Http1ResponseDecoder.java

```diff
@@ -194,10 +202,30 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
                         assert res != null;
                         this.res = res;
 
-                        res.startResponse();
                         final ResponseHeaders responseHeaders = ArmeriaHttpUtil.toArmeria(nettyRes);
+                        final HttpStatus status = responseHeaders.status();
+
+                        if (res.needs100Continue()) {
+                            if (status == HttpStatus.CONTINUE) {
+                                res.resume();
+                            } else if (status == HttpStatus.EXPECTATION_FAILED) {
+                                // Remove the response and do not close the HttpResponseWrapper in order not to
+                                // close the original response.
+                                removeResponse(resId++);
+                                // repeat can't fail for HTTP/1.1 because it doesn't use
+                                // settings frame to reduce the number of max concurrent streams.
```

Comment:
e.g. Actually, I think if `KeepAliveHandler` tries to disconnect the connection this can be false.
ref: https://github.com/line/armeria/blob/5998bb59975cfc9df9a3b1ac3c7ef801ac1d9ab1/core/src/main/java/com/linecorp/armeria/client/AbstractHttpRequestHandler.java#L189

```suggestion
```

Reply:
Oops, thanks a lot for finding this. Let me fix that soon. 😄